tests/test_algorithm.py

```diff
@@ -3531,6 +3532,42 @@ def handle_data(algo, data):
                                        env=self.env)
         self.check_algo_succeeds(algo, handle_data)
 
+    def test_set_min_leverage(self):
+        # Algo date is not past deadline so algorithm succeeds
+        offset = pd.Timedelta('10 days')
+
+        # Set min leverage to 0 so buying one share fails.
+        def handle_data(algo, data):
+            algo.order(algo.sid(self.sidint), 1)
+            algo.record(latest_time=algo.get_datetime())
+
+        algo = SetMinLeverageAlgorithm(1, offset, sim_params=self.sim_params,
+                                       env=self.env)
+        self.check_algo_succeeds(algo, handle_data)
+
+        # Algo date is past deadline
+        # Set min leverage to 1 so buying one share fails.
+        offset = pd.Timedelta('1 day')
+
+        def handle_data(algo, data):
+            algo.order(algo.sid(self.sidint), 1)
+            algo.record(latest_time=algo.get_datetime())
+
+        algo = SetMinLeverageAlgorithm(1, offset, sim_params=self.sim_params,
+                                       env=self.env)
+        self.check_algo_fails(algo, handle_data)
+        self.assertEqual(
+            algo.recorded_vars['latest_time'],
+            pd.Timestamp('2006-01-04 21:00:00', tz='UTC'),
+        )
+
+        # Set min leverage to 0 so buying one share passes
+        def handle_data(algo, data):
+            algo.order(algo.sid(self.sidint), 1)
+
+        algo = SetMinLeverageAlgorithm(0, offset, sim_params=self.sim_params,
+                                       env=self.env)
+        self.check_algo_succeeds(algo, handle_data)
```

Comment:
I think we need a test case where we set a minimum leverage amount and the algorithm has a leverage above that limit after the deadline and doesn't crash.  Bonus points if the algorithm then goes below the leverage limit and crashes.

Reply:
This might be easier to implement using `order_target_percent` to hit fixed leverage amounts instead of ordering fixed numbers of shares.